08_AdvancedTypes/UniversalTablePrinter.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Reflection;
+using System.Text;
+
+public class UniversalTablePrinter
+{
+    private readonly IEnumerable<object> _data;
+    private const int _columnLength = 20;
+
+    public UniversalTablePrinter(IEnumerable<object> data)
+    {
+        _data = data;
+    }
+    public void PrintToConsole()
+    {
+        object firstElement = _data.First();
+        if (firstElement is not null)
+        {
+            Type type = firstElement.GetType();
+            PropertyInfo[] props = type.GetProperties();
+
+            //add table columns name
+            StringBuilder columns = new StringBuilder();
+            foreach (PropertyInfo prop in props)
+            {
+                columns.Append($"{prop.Name,-_columnLength}|");
+            }
+            columns.AppendLine();
+            columns.Append(new string('-', (_columnLength + 1) * props.Length));
+            Console.WriteLine(columns);
+
+            //add the data
+            StringBuilder rows = new StringBuilder();
```

Comment:
Why have a separate `StringBuilder` for columns and rows? You could use the same